### PR TITLE
Indicate that the type pattern supports generic arguments

### LINKED ISSUE #10334: Missing introduction of pattern matching on generic type parameters
This is valid in C# 7.1 but not in C# 7.0:

```
public void Method<T>(T param)
{
    switch (param)
    {
        case A a:
            Console.WriteLine("A");
            break;
        case B b:
            Console.WriteLine("B");
            break;
    }
}
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6d38d8df-4919-e8c7-8126-e8c5c7300523
* Version Independent ID: b2284f21-547d-01f5-76a3-b531aab1a909
* Content: [What's new in C# 7.1](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-1)
* Content Source: [docs/csharp/whats-new/csharp-7-1.md](https://github.com/dotnet/docs/blob/master/docs/csharp/whats-new/csharp-7-1.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #10334

Relies on dotnet/samples#803

In addition, review and update modified topics.

### CODE REVIEW

**Comment by BillWagner** [comment]
closing and re-opening for a fresh build.

File: docs/csharp/language-reference/keywords/is.md [hunk 1]
```diff
@@ -63,15 +63,15 @@ Starting with C# 7.0, the `is` and [switch](../../../csharp/language-reference/k
 
 ### <a name="type" /> Type pattern </a>
```
Comment (Thraka) [inline docs/csharp/language-reference/keywords/is.md]: nit: I just noticed this, not related to your changes though. The anchor link terminates in a single statement and then follows the heading with another termination. I think the `</a>` gets stripped out though

File: docs/csharp/language-reference/keywords/switch.md [hunk 2]
```diff
@@ -167,6 +167,12 @@ The following example uses the type pattern to provide information about various
 
 [!code-csharp[type-pattern#1](~/samples/snippets/csharp/language-reference/keywords/switch/type-pattern.cs#1)]
 
+Instead of `object`, you could make a generic method, using the type of the collection as the type parameter, as shown in the following code:
+
+[!code-csharp[type-pattern#1](~/samples/snippets/csharp/language-reference/keywords/switch/type-pattern3.cs#3)]
```
Comment (Thraka) [inline docs/csharp/language-reference/keywords/switch.md]: This is blank. Is the `#3` at the end of the file name supposed to be `#1`?

**Review by Thraka — APPROVED** [review]
The only glaring problem is that I believe the code example reference is wrong.

#### Is keyword
Unrelated to your changes: The **var pattern** section is formed differently from the others. The explanation about what this pattern does is after the code example. I found myself digging into the code and the intro paragraph trying to understand why I would ever use this; it seemed pointless. The single sentence after the example cleared that up. You should lead with that info :)

**Pull request merged**